asteroid/utils/torch_utils.py

```diff
@@ -123,3 +124,87 @@ def are_models_equal(model1, model2):
         if p1.data.ne(p2.data).sum() > 0:
             return False
     return True
+
+class OverlapAddWrapper(torch.nn.Module):
+    def __init__(self, nnet, sources, window_size, device, window="hanning", reorder_chunks=True):
+        super(OverlapAddWrapper, self).__init__()
+
+        assert window_size % 2 == 0, "Window size must be even"
+
+        self.to(device)
+        self.device = device
+        if isinstance(nnet, torch.nn.Module):
+            nnet = nnet.to(device)
```

Comment:
I don't understand why we need that part.

Reply:
in case nnet is a function for whatever reason one might want to apply a function window by window

Reply:
But we can do this at runtime right? Send the Module to the device of the input for example? Or is it better to specify the device beforehand. 
I would treat it like a normal `nn.Module`

Reply:
yeah you are right. Do we also need the whole to be differentiable  ?